src/Package/Fetch.zig

```diff
@@ -608,6 +630,16 @@ fn loadManifest(f: *Fetch, pkg_root: Package.Path) RunError!void {
     }
 }
 
+fn archivePackageDir(arena: Allocator, out_dir: fs.Dir) !?[]const u8 {
+    var iter = out_dir.iterate();
+    if (try iter.next()) |entry| {
+        if (try iter.next() == null and entry.kind == .directory) {
+            return try arena.dupe(u8, entry.name);
+        }
+    }
+    return null;
+}
```

Comment:
This information should be instead tracked by `pipeToFileSystem` to avoid these two syscalls (`open` and `readdir`).